uxarray/grid/coordinates.py

```diff
@@ -244,6 +209,186 @@ def _construct_face_centroids(node_x, node_y, node_z, face_nodes, n_nodes_per_fa
     return _normalize_xyz(centroid_x, centroid_y, centroid_z)
 
 
+def circle_from_two_points(p1, p2):
+    """Calculate the smallest circle that encloses two points on a unit sphere.
+
+    Parameters
+    ----------
+    p1 : tuple
+        The first point as a tuple of (latitude, longitude).
+    p2 : tuple
+        The second point as a tuple of (latitude, longitude).
+
+    Returns
+    -------
+    tuple
+        A tuple containing the center (as a tuple of lon and lat) and the radius of the circle.
+    """
+    center_lon = (p1[0] + p2[0]) / 2
+    center_lat = (p1[1] + p2[1]) / 2
+    center = (center_lon, center_lat)
+    v1, v2 = (np.array(_lonlat_rad_to_xyz(*np.radians(p))) for p in (p1, p2))
+    distance = _angle_of_2_vectors(v1, v2)
+    radius = distance / 2
```

Comment:
Is the radius the distance on the sphere, or the distance on a plane?

Reply:
> Is the radius the distance on the sphere, or the distance on a plane?

welzl is typically for a set of points doesn't have to be on a sphere, the algorithm doesn't force anything on the points it works with. Here, I'm making it work as if on a sphere and calculating the distances on sphere instead of a plane, so we just need Haversine distance, I check that `_angle_of_2_vectors` is same as Haversine distance.

Reply:
When comes to the spherical surface, you can use the modified version called [Bounding Sphere](https://en.wikipedia.org/wiki/Bounding_sphere), it's the 3-D version of the welzl's algorithmn

Reply:
> When comes to the spherical surface, you can use the modified version called [Bounding Sphere](https://en.wikipedia.org/wiki/Bounding_sphere), it's the 3-D version of the welzl's algorithmn

Hey, I was thinking about this and was half way through my implementation - made a rookie mistake of not thinking through before implentation --

 the current algorithm is correct - it is modified for 2d spherical geometry and the distance is computed using the great circle distance. We don't need 3D sphere for this.